pkg/antctl/raw/multicluster/common/common.go

```diff
@@ -0,0 +1,355 @@
+	if saName == "" {
+		saName = name
+	}
+	if rbName == "" {
+		rbName = name
+	}
+
+	var createErr error
+	serviceAccount := newServiceAccount(saName, namespace)
+	unstructuredSA, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(serviceAccount)
+
+	createErr = k8sClient.Create(context.TODO(), serviceAccount)
+	if createErr != nil {
+		if !apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create ServiceAccount \"%s\", error: %s\n", saName, createErr.Error())
+			return createErr
+		}
+		fmt.Fprintf(cmd.OutOrStderr(), "ServiceAccount \"%s\" already exists\n", saName)
+		createErr = nil
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "ServiceAccount \"%s\" created\n", serviceAccount.Name)
+		*createdRes = append(*createdRes, unstructuredSA)
+	}
+
+	roleBinding := newRoleBinding(rbName, saName, namespace)
+	unstructuredRoleBinding, _ := runtime.DefaultUnstructuredConverter.ToUnstructured(roleBinding)
+
+	createErr = k8sClient.Create(context.TODO(), roleBinding)
+	if createErr != nil {
+		if !apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create RoleBingding \"%s\", error: %s\n", rbName, createErr.Error())
+			return createErr
+		}
+		fmt.Fprintf(cmd.OutOrStderr(), "RoleBinding \"%s\" already exists\n", rbName)
+		createErr = nil
+	} else {
+		fmt.Fprintf(cmd.OutOrStdout(), "RoleBinding \"%s\" created\n", roleBinding.Name)
+		*createdRes = append(*createdRes, unstructuredRoleBinding)
+	}
+
+	secret := newSecret(name, saName, namespace)
+	createErr = k8sClient.Create(context.TODO(), secret)
+	if createErr != nil {
+		if !apierrors.IsAlreadyExists(createErr) {
+			fmt.Fprintf(cmd.OutOrStdout(), "Failed to create Secret \"%s\", start rollback\n", name)
+			return createErr
+		}
+		fmt.Fprintf(cmd.OutOrStderr(), "Secret \"%s\" already exists\n", name)
+	}
+	// It will take one or two seconds to wait for the Data.token to be created.
+	if err := waitForSecretReady(k8sClient, name, namespace); err != nil {
+		return err
+	}
+
+	if file == nil {
+		return nil
+	}
+
+	if err := k8sClient.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: namespace}, secret); err != nil {
+		return err
+	}
+	s := &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "v1",
+			Kind:       "Secret",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: name,
+		},
+		Data: secret.Data,
+		Type: corev1.SecretTypeOpaque,
+	}
+
+	b, err := yaml.Marshal(s)
+	if err != nil {
+		return err
+	}
+	if _, err := file.Write([]byte("---\n")); err != nil {
+		return err
+	}
+	if _, err := file.Write(b); err != nil {
+		return err
+	}
+	fmt.Fprintf(cmd.OutOrStdout(), "Member token saved to %s\n", file.Name())
+
+	return nil
+}
+
+func waitForSecretReady(client client.Client, secretName string, namespace string) error {
+	return wait.PollImmediate(
+		1*time.Second,
+		30*time.Second,
```

Comment:
If it's 1 or 2 seconds to wait Secret, maybe change 30 to 5 is enough, otherwise, the command will stuck here for a quite long time.

Reply:
Changed the wait time to 5 seconds.